src/storage/meta/meta_map.cppm

```diff
@@ -14,51 +14,135 @@
+                                                                          TxnManager *txn_mgr) {
+    std::shared_lock r_lock(rw_locker_);
+    auto iter = meta_map_.find(name);
+    if (iter == meta_map_.end()) {
+        r_lock.unlock();
+        {
+            UniquePtr<Meta> new_meta = init_func();
+            std::unique_lock w_lock(rw_locker_);
+            if (auto iter1 = meta_map_.find(name); iter1 != meta_map_.end()) {
+                LOG_TRACE("Add new entry in new meta_map");
+                iter = iter1;
+            } else {
+                LOG_TRACE("Add new entry in existed meta_map");
+                // When replay database creation,if the txn_manger is nullptr, represent the txn manager not running, it is reasonable.
+                // In replay phase, not need to recording the catalog delta operation.
+                if (txn_mgr != nullptr) {
+                    auto operation = MakeUnique<typename Meta::MetaOp>(new_meta.get(), begin_ts);
+                    txn_mgr->GetTxn(txn_id)->AddCatalogDeltaOperation(std::move(operation));
+                }
+                iter = meta_map_.emplace(name, std::move(new_meta)).first;
+            }
+        }
+        r_lock.lock();
+    } else {
+        LOG_TRACE("Add new entry in existed meta_map");
+    }
+    return {iter->second.get(), std::move(r_lock)};
+}
+
+template <MetaConcept Meta>
+Tuple<Meta *, Status, std::shared_lock<std::shared_mutex>> MetaMap<Meta>::GetExistMeta(const String &name, ConflictType conflict_type) {
+    std::shared_lock r_lock(rw_locker_);
+    if (auto iter = meta_map_.find(name); iter != meta_map_.end()) {
+        return {iter->second.get(), Status::OK(), std::move(r_lock)};
+    }
+    if (conflict_type == ConflictType::kIgnore) {
+        LOG_TRACE(fmt::format("Ignore drop a non-exist meta: {}", name));
+        return {nullptr, Status::OK(), std::move(r_lock)};
+    }
+    auto err_msg = MakeUnique<String>(fmt::format("Attempt to drop non-exist entry {}", name));
+    LOG_ERROR(*err_msg);
+    if constexpr (std::is_same_v<Meta, TableMeta>) {
+        return {nullptr, Status(ErrorCode::kTableNotExist, std::move(err_msg)), std::move(r_lock)};
+    } else if constexpr (std::is_same_v<Meta, DBMeta>) {
+        return {nullptr, Status(ErrorCode::kDBNotExist, std::move(err_msg)), std::move(r_lock)};
+    } else if constexpr (std::is_same_v<Meta, TableIndexMeta>) {
+        return {nullptr, Status(ErrorCode::kIndexNotExist, std::move(err_msg)), std::move(r_lock)};
+    } else {
+        UnrecoverableError("Unimplemented");
+    }
+}
+
+template <MetaConcept Meta>
+void MetaMap<Meta>::Iterate(std::function<void(Meta *)> func) {
     std::unique_lock lock(rw_locker_);
+    for (auto &[name, meta] : meta_map_) {
+        func(meta.get());
+    }
+}
 
+template <MetaConcept Meta>
+void MetaMap<Meta>::PickCleanup(CleanupScanner *scanner) {
+    std::unique_lock w_lock(rw_locker_);
     for (auto iter = meta_map_.begin(); iter != meta_map_.end();) {
```

Comment:
1. Collect all meta_map value ptr, and save it to vector
2. Loop the vector to pick entry to clean up.